proof_of_concept/asset.py

```diff
@@ -1,58 +1,66 @@
 """Classes for describing assets."""
 from typing import Any, Dict, Optional
 
+from proof_of_concept.swagger.base_model_ import Model
 from proof_of_concept.workflow import Job
 
 
-class Metadata:
+class Metadata(Model):
     """Stores metadata for stored assets.
-
-    Attributes:
-        job (Job): A minimal job that will generate this asset.
-        item (str): The item in the job's workflow corresponding to
-                this asset.
-
     """
+    swagger_types = {
+        'job': Job,
+        'item': str
+    }
 
-    def __init__(self, job: Job, item: str) -> None:
-        """Create a Metadata object.
+    attribute_map = {
+        'job': 'job',
+        'item': 'item'
+    }
 
-        Args:
-            job: A minimal job that will generate this asset.
-            item: The item in the job's workflow corresponding to this
-                    asset.
+    def __init__(self, job: Job, item: str):
+        """Create a metadata object.
 
+        Args:
+            job (Job): A minimal job that will generate this asset.
+            item (str): The item in the job's workflow corresponding to
+                    this asset.
         """
         self.job = job
         self.item = item
 
 
-class Asset:
-    """Asset, a representation of a computation or piece of data."""
+class Asset(Model):
```

Comment:
I decided to just stick to Asset and not care about whether it is a data or compute asset. This seems to work flawlessly, not sure if it adheres to theoretical concepts, @LourensVeen ?